template/template_test.go

```diff
@@ -1606,8 +1607,87 @@ func TestTemplate_Execute(t *testing.T) {
 			"1",
 			false,
 		},
+		{
+			"leaf_cert",
+			&NewTemplateInput{
+				Contents: `{{with caLeaf "foo"}}` +
+					`{{.CertPEM}}{{.PrivateKeyPEM}}{{end}}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					d := dep.NewConnectLeafQuery("foo")
+					b := NewBrain()
+					b.Remember(d, &api.LeafCert{
+						Service:       "foo",
+						CertPEM:       "PEM",
+						PrivateKeyPEM: "KEY",
+					})
+					return b
+				}(),
+			},
+			"PEMKEY",
+			false,
+		},
+		{
+			"root_ca",
+			&NewTemplateInput{
+				Contents: `{{range caRoots}}{{.RootCertPEM}}{{end}}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					d := dep.NewConnectCAQuery()
+					b := NewBrain()
+					b.Remember(d, []*api.CARoot{
+						&api.CARoot{
+							Name:        "Consul CA Root Cert",
+							RootCertPEM: "PEM",
+							Active:      true,
+						},
+					})
+					return b
+				}(),
+			},
+			"PEM",
+			false,
+		},
+		{
+			"func_connect",
+			&NewTemplateInput{
+				Contents: `{{ range connect "webapp" }}{{ .Address }}{{ end }}`,
+			},
+			&ExecuteInput{
+				Brain: func() *Brain {
+					b := NewBrain()
+					d, err := dep.NewHealthServiceQuery("webapp")
+					if err != nil {
+						t.Fatal(err)
+					}
+					b.Remember(d, []*dep.HealthService{
+						&dep.HealthService{
+							Node:    "node1",
+							Address: "1.2.3.4",
+						},
+						&dep.HealthService{
+							Node:    "node2",
+							Address: "5.6.7.8",
+						},
+					})
+					return b
+				}(),
+			},
+			"1.2.3.45.6.7.8",
+			false,
+		},
 	}
 
+	//	struct {
```

Comment:
stray notes?

Reply:
Sort of... notes that I thought I'd keep. I'm always referring to the struct type at the very top when adding tests at the bottom. Annoying to flip back and forth. Thought I'd just add a commented copy of the struct where I could see it.